src/utils/validation.ts

```diff
@@ -0,0 +1,65 @@
+import { providerErrors, rpcErrors } from '@metamask/rpc-errors';
+import type { Struct, StructError } from '@metamask/superstruct';
+import { validate } from '@metamask/superstruct';
+import type { Hex, JsonRpcRequest } from '@metamask/utils';
+
+export async function validateAndNormalizeKeyholder(
+  address: Hex,
+  req: JsonRpcRequest,
+  { getAccounts }: { getAccounts: (req: JsonRpcRequest) => Promise<string[]> },
+): Promise<Hex> {
+  if (
+    typeof address === 'string' &&
+    address.length > 0 &&
+    resemblesAddress(address)
+  ) {
```

Comment:
is there some helper in utils we could use here?

Reply:
ahh i see. you're just DRYing this helper up. ehh okay, let's just leave as is then

Reply:
Exactly, I wanted to use `isValidHexAddress` but didn't want to impact all the existing usages.